gossip/poset_hook.go

```diff
@@ -60,7 +60,7 @@ func (hook *HookedEngine) GetEpoch() idx.Epoch {
 // GetEpochValidators atomically returns validators of current epoch, and the epoch.
 func (hook *HookedEngine) GetEpochValidators() (*pos.Validators, idx.Epoch) {
 	if hook.engine == nil {
-		return pos.NewValidators(), 1
+		return pos.NewBuilder().Build(), 1
```

Comment:
check hook.engine, if not initialized, then initialize it.

if hook.engine == nil {
  hook.engine = ...
}

HookedEngine has 'engine' field? delegation call? 
line 65: risk of infinite loop.

Reply:
Yes, HookedEngine is a decorator for engine, which overloads ProcessEvent Method and also returns empty values if engine is nil (not currently used, but nil engine may be used in test in future)